### ISSUE TITLE
General: Passthrough default props and icons

### ISSUE THREAD

**Issue by m-thorsen** [primefaces/primevue#4607]
### Describe the feature you would like to see added

Add possibility to define icons and props via passthrough in order to avoid having to wrap/extend/template component every time.  The passthrough option seems like such an ideal place for  this!

Some examples:
- I always change the Panel collapse icon to angle - similar to accordion
- I always set dialog defaults draggable=false, modal=true
- I always set Select/Multiselect props auto-filter-focus=true, scroll-height=<larger> 

### Describe the solution you'd like

Example: 

```vue
<Panel :pt="{
   togglerIcon: ({ props}) => ({
      icon: props.collapsed ? 'pi-angle-up' : 'pi-angle-down'
   })
}" />
```

AND/OR

```vue
import { Icon } from '@iconify/vue';
...
<Panel :pt="{
   togglerIcon: ({ props}) => ({
      icon:  `<Icon icon=${props.collapsed ? 'mdi:chevron-up'  :  'mdi:chevron-down'} />`
   })
}" />
```

---

```vue
<Dialog :pt="{
   root: {
      modal: true,
      draggable: false,
   }
}" />
```

OR

```vue
<Dialog :pt="{
   modal: true,
   draggable: false,
}" />
```

### Describe alternatives you have considered

Wrapping each component, templating every use case

**Comment by mertsincan** [comment]
Hi @m-thorsen,

I did a preliminary study on this. https://twitter.com/primevue/status/1656258065014968324
Of course, I will try to create an API related to it and make it possible to do this on a component basis. But, in future. Now, I'm working on a new styled mode structure. After completing this, I'll start the default options configuration.

Best Regards,

**Comment by slavco86** [comment]
Faced the same problem as well. Lack of ability to pass through prop values down to internal components of more complex components. E.g. passing props to Button inside DataTable, such as `severity`, `label`.
Any  update on this, @mertsincan ?
Is this something we can expect to see in V4?

**Comment by alqa-dash** [comment]
Interesting thread. I tried to change Panel toggle icon but with example above it does not work. At least in v.3.51.0.
`<Panel :pt="{
   togglerIcon: ({ props}) => ({
      icon: props.collapsed ? 'pi-angle-up' : 'pi-angle-down'
   })
}" />`

I manage to change icon with <template #togglericon> but want to know is it possible with passthrough approach?

Looks like passthrough options are pretty powerful.

**Comment by mostafaznv** [comment]
> Hi @m-thorsen,
> 
> I did a preliminary study on this. https://twitter.com/primevue/status/1656258065014968324 Certainly, I plan to create an API related to it and enable this functionality on a component basis, but that will be in the future. Right now, I am working on developing a new styled mode structure. Once I complete this task, I will move on to configuring the default options.
> 
> Best Regards,

Do we have this feature in v4?
I mean, setting default values for components using global configuration

**Comment by guillaume-bonnefoy** [comment]
Interested in this too

**Comment by offbit-it** [comment]
+1

**Comment by anchan42** [comment]
This would be really nice and solve a lot of problem.

**Comment by pindab0ter** [comment]
> Hi [@m-thorsen](https://github.com/m-thorsen),
> 
> I did a preliminary study on this. https://twitter.com/primevue/status/1656258065014968324 Certainly, I plan to create an API related to it and enable this functionality on a component basis, but that will be in the future. Right now, I am working on developing a new styled mode structure. Once I complete this task, I will move on to configuring the default options.
> 
> Best Regards,

@mertsincan Any news on this?

**Comment by albertjannap** [comment]
@pindab0ter if you found one, what was your work around for this issue?

**Comment by mostafaznv** [comment]
> [@pindab0ter](https://github.com/pindab0ter) if you found one, what was your work around for this issue?

I decided to move on. 

NuxtUI and RekaUI offer better features and functionalities for customizing and maintaining projects.
PrimeVue is also a good option, but its lack of focus on a single project—along with the need to maintain libraries for React, Angular, and other JavaScript frameworks—has slowed its progress.
There is still a lot of work to be done, but the team refuses to acknowledge this. They believe their project is at its peak and doesn’t require any improvements or changes, which is unfortunate.

**Comment by pindab0ter** [comment]
> [@pindab0ter](https://github.com/pindab0ter) if you found one, what was your work around for this issue?

The only workaround that I can think of is to define your own component using [Fallthrough Attributes](https://vuejs.org/guide/components/attrs.html) and use that instead.

**Comment by blouflashdb** [comment]
Really sad to see that its still not supported. I guess I am going to switch to NuxtUI.

**Comment by guillaume-bonnefoy** [comment]
Don't act so entitled, this is free and open source. No one owes you anything.
You can contribute if you're not happy.

**Comment by anchan42** [comment]
I don't see how any of the comments here can be intepreted as being entitled, to be honest. Being free and open source does not mean it is above any criticism. On the contrary, constructive  criticism should be invited. Not all people have time or technical expertise to contribute. There are number of issues that are clearly a problem from user's perspective but get closed with reason like that the way it is supposed to be. Lacking of focus is something worth considering about.

**Comment by imikho1ls** [comment]
+1

**Comment by iamvanja** [comment]
Any update on this, @mertsincan? The [X post](https://x.com/primevue/status/1656258065014968324) looks super exciting!

**Comment by patrick-kennedy-fctg** [comment]
Coming to PrimeVue for a new job and I have to say this was such a huge blessing in Vuetify, especially being able to define default settings in a hierarchy (e.g. a Button in a Dialog having different defaults than a Button alone). Being required to wrap every component that needs defaults is not just a chore but introduces a large amount of cruff that has to be managed by the team.

tl;dr this would be super helpful!
